File: android/src/main/java/com/vidyoplatform/connector/plugin/event/IPluginEventHandler.java
package com.vidyoplatform.connector.plugin.event;

import com.vidyo.VidyoClient.Endpoint.Participant;

public interface IPluginEventHandler {

    void onInitialized();

    void onConnected();

    void onDisconnected(String reason);

    void onFailure(String reason);

    void onParticipantJoined(Participant participant);

    void onParticipantLeft(Participant participant);
}

File: ios/Plugin/IPluginEventHandler.swift
//
//  IPluginEventHandler.swift
//  Plugin
//

import Foundation

protocol IPluginEventHandler: class {

    func onInitialized(status: Bool)

    func onConnected()

    func onDisconnected(reason: String)

    func onFailure(reason: String)

    func onParticipantJoined(participant: VCParticipant)

    func onParticipantLeft(participant: VCParticipant)
}
